### PR TITLE
Add test case to check file name and content type for `mulipart/form-data`

### PR DESCRIPTION
**Pull request by msyyc**
fixes https://github.com/Azure/cadl-ranch/issues/502

### CODE REVIEW

**Review by MaryGao — APPROVED** [review]
LGTM

**Comment by msyyc** [comment]
test it locally:
```python
import requests

files = (
    ('profileImage',
     ("image.jpg", open("D:\\dev1\\cadl-ranch\\packages\\cadl-ranch-specs\\assets\\image.jpg", "rb"), "image/jpg")),
)
requests.post(url="http://127.0.0.1:3000/multipart/form-data/check-filename-and-content-type", files=files,
              data={"id": "123"})
```

File: packages/cadl-ranch-specs/cadl-ranch-summary.md [hunk 1]
```diff
@@ -1667,6 +1667,30 @@ Content-Type: application/octet-stream
...
+this case will check filename and content-type of file part, so expect request:
+
+```
+POST /upload HTTP/1.1
+Content-Length: 428
+Content-Type: multipart/form-data; boundary=abcde12345
+
+--abcde12345
+Content-Disposition: form-data; name="id"
+Content-Type: text/plain
```
Comment (weidongxu-microsoft) [inline packages/cadl-ranch-specs/cadl-ranch-summary.md:1683]: nit, why we put an "id" into "check-filename-and-content-type"

Comment (msyyc) [reply packages/cadl-ranch-specs/cadl-ranch-summary.md:1683]: I just want to reuse the existing model...

**Review by weidongxu-microsoft — APPROVED** [review]

**Pull request merged**